### DISCUSSION TITLE
Error when trying to train ljspeech dataset using tacotron2-DCA

### DISCUSSION THREAD

**Discussion by snufas** [coqui-ai/TTS#1937] [General Q&A]
Hello folks, I would highly appreciate your input regarding this issue, Keep in mind that I am a newbie, and I don't have any clue what the errors below are actually indicating.

The output below:

(coqui-aiTTS) root@DESKTOP-7A6UGRU:/home/snufas/github_projects/coqui-aiTTS/training/tacotron2-DCA# CUDA_VISIBLE_DEVICES="0" python train_tacotron_dca.py
 > Setting up Audio Processor...
 | > sample_rate:22050
 | > resample:False
 | > num_mels:80
 | > log_func:np.log
 | > min_level_db:-100
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:20
 | > fft_size:1024
 | > power:1.5
 | > preemphasis:0.0
 | > griffin_lim_iters:60
 | > signal_norm:False
 | > symmetric_norm:True
 | > mel_fmin:0
 | > mel_fmax:8000
 | > pitch_fmin:0.0
 | > pitch_fmax:640.0
 | > spec_gain:1.0
 | > stft_pad_mode:reflect
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:True
 | > trim_db:60.0
 | > do_sound_norm:False
 | > do_amp_to_db_linear:True
 | > do_amp_to_db_mel:True
 | > do_rms_norm:False
 | > db_level:None
 | > stats_path:None
 | > base:2.718281828459045
 | > hop_length:256
 | > win_length:1024
 | > Found 13100 files in /home/snufas/github_projects/coqui-aiTTS/training/LJSpeech-1.1
 > Using CUDA: True
 > Number of GPUs: 1

 > Model has 28259417 parameters

 > EPOCH: 0/1000
 --> /home/snufas/github_projects/coqui-aiTTS/training/tacotron2-DCA/run-September-06-2022_04+57PM-096b35f6
[*] Pre-computing phonemes...
  1%|▍                                                                               | 74/12969 [00:04<12:39, 16.97it/s]ðə kəmˈɪʃən bᵻlˈiːvz ðætðə fˈækts ʌvðɪ ɐsˌæsᵻnˈeɪʃən ʌv pɹˈɛzɪdənt kˈɛnədi pˈɔɪnt tə sˈɜːʔn̩ mˈɛʒɚz wˈɪtʃ,
 [!] Character '̩' not found in the vocabulary. Discarding it.
 16%|████████████▎                                                                 | 2054/12969 [02:47<13:09, 13.82it/s]ˈɪntʊ ðə “kɹˈeɪɾɚ” dˈʌɡ ˈaʊt ɪnðə mˈɪdəl, pˈoːɹ ðə spˈʌndʒ, wˈɔːɹm wˈɔːɾɚ, ðə məlˈæsᵻz, ænd sˈoʊdə dɪsˈɑːlvd ɪn hˈɑːt wˈɔːɾɚ.
 [!] Character '“' not found in the vocabulary. Discarding it.
ˈɪntʊ ðə “kɹˈeɪɾɚ” dˈʌɡ ˈaʊt ɪnðə mˈɪdəl, pˈoːɹ ðə spˈʌndʒ, wˈɔːɹm wˈɔːɾɚ, ðə məlˈæsᵻz, ænd sˈoʊdə dɪsˈɑːlvd ɪn hˈɑːt wˈɔːɾɚ.
 [!] Character '”' not found in the vocabulary. Discarding it.
100%|█████████████████████████████████████████████████████████████████████████████| 12969/12969 [20:39<00:00, 10.46it/s]

> DataLoader initialization
| > Tokenizer:
        | > add_blank: False
        | > use_eos_bos: False
        | > use_phonemes: True
        | > phonemizer:
                | > phoneme language: en-us
                | > phoneme backend: espeak
        | > 3 not found characters:
        | > ̩
        | > “
        | > ”
| > Number of instances : 12969
 | > Preprocessing samples
 | > Max text length: 188
 | > Min text length: 13
 | > Avg text length: 100.90014650319993
 |
 | > Max audio length: 222643.0
 | > Min audio length: 24499.0
 | > Avg audio length: 144984.29755570978
 | > Num. instances discarded samples: 0
 | > Batch group size: 0.

 > TRAINING (2022-09-06 17:18:17)
Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.
Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.
Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.
Numba: Attempted to fork from a non-main thread, the TBB library may be in an invalid state in the child process.
/home/snufas/github_projects/coqui-aiTTS/TTS/tts/models/tacotron2.py:331: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  alignment_lengths = (

   --> STEP: 0/203 -- GLOBAL_STEP: 0
     | > decoder_loss: 32.89396  (32.89396)
     | > postnet_loss: 34.96359  (34.96359)
     | > stopnet_loss: 0.65256  (0.65256)
     | > loss: 17.61695  (17.61695)
     | > align_error: 0.91468  (0.91468)
     | > grad_norm: 3.26834  (3.26834)
     | > current_lr: 0.00000
     | > step_time: 2.55640  (2.55644)
     | > loader_time: 0.65560  (0.65559)

/home/snufas/github_projects/coqui-aiTTS/TTS/tts/models/tacotron2.py:335: UserWarning: __floordiv__ is deprecated, and its behavior will change in a future version of pytorch. It currently rounds toward 0 (like the 'trunc' function NOT 'floor'). This results in incorrect rounding for negative values. To keep the current behavior, use torch.div(a, b, rounding_mode='trunc'), or for actual floor division, use torch.div(a, b, rounding_mode='floor').
  alignment_lengths = mel_lengths // self.decoder.r

   --> STEP: 25/203 -- GLOBAL_STEP: 25
     | > decoder_loss: 34.04894  (33.39444)
     | > postnet_loss: 36.04324  (35.40215)
     | > stopnet_loss: 0.65330  (0.65150)
     | > loss: 18.17635  (17.85065)
     | > align_error: 0.95518  (0.94287)
     | > grad_norm: 2.94930  (2.93635)
     | > current_lr: 0.00000
     | > step_time: 1.81480  (1.24027)
     | > loader_time: 0.00560  (0.00519)

 ! Run is removed from /home/snufas/github_projects/coqui-aiTTS/training/tacotron2-DCA/run-September-06-2022_04+57PM-096b35f6
Traceback (most recent call last):
  File "/root/miniconda3/envs/coqui-aiTTS/lib/python3.9/site-packages/trainer/trainer.py", line 1533, in fit
    self._fit()
  File "/root/miniconda3/envs/coqui-aiTTS/lib/python3.9/site-packages/trainer/trainer.py", line 1517, in _fit
    self.train_epoch()
  File "/root/miniconda3/envs/coqui-aiTTS/lib/python3.9/site-packages/trainer/trainer.py", line 1282, in train_epoch
    _, _ = self.train_step(batch, batch_num_steps, cur_step, loader_start_time)
  File "/root/miniconda3/envs/coqui-aiTTS/lib/python3.9/site-packages/trainer/trainer.py", line 1114, in train_step
    outputs, loss_dict_new, step_time = self._optimize(
  File "/root/miniconda3/envs/coqui-aiTTS/lib/python3.9/site-packages/trainer/trainer.py", line 1050, in _optimize
    loss_dict["loss"].backward()
  File "/root/miniconda3/envs/coqui-aiTTS/lib/python3.9/site-packages/torch/_tensor.py", line 396, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/root/miniconda3/envs/coqui-aiTTS/lib/python3.9/site-packages/torch/autograd/__init__.py", line 173, in backward
    Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: CUDA error: unknown error
CUDA kernel errors might be asynchronously reported at some other API call,so the stacktrace below might be incorrect.
For debugging consider passing CUDA_LAUNCH_BLOCKING=1.
(coqui-aiTTS) root@DESKTOP-7A6UGRU:/home/snufas/github_projects/coqui-aiTTS/training/tacotron2-DCA#

**Comment by erogol** [comment]
I think it is about insufficient GPU memory. Try a larger memory or reduce batchsize
